userapi/src/main/java/model/IDataModel.java

```diff
@@ -0,0 +1,57 @@
+package model;
+
+import fk.prof.storage.AsyncStorage;
+
+import java.util.Set;
+
+/**
+ * Interface for DataStores containing aggregated profile data
+ * Created by rohit.patiyal on 23/01/17.
+ */
+public interface IDataModel {
+    /**
+     * Initializes the DataModel with DataStorage {@link AsyncStorage}
+     *
+     * @param store datastore object
+     */
+    void setStorage(AsyncStorage store);
```

Comment:
Removed the setter method. Now it is required by the implementer of the interface to instantiate the `ProfileDiscoveryAPI` using a constructor with an `AsyncStorage` parameter.